src/indieweb_utils/indieauth/server.py

```diff
@@ -0,0 +1,333 @@
+import time
+import hashlib
+import base64
+import binascii
+import os
+import jwt
+
+from dataclasses import dataclass
+
+
+def generate_token(*, size: int = 20) -> str:
+    return binascii.hexlify(os.urandom(size)).decode()
+
+
+class AuthenticationError(Exception):
+    pass
+
+
+class TokenValidationError(Exception):
+    pass
+
+
+class AuthorizationCodeExpiredError(Exception):
+    pass
+
+
+@dataclass
+class DecodedAuthToken:
+    me: str
+    client_id: str
+    scope: str
+    decoded_authorization_code: dict
+
+
+@dataclass
+class AuthTokenResponse:
+    code: str
+    code_verifier: str
+    code_challenge: str
+
+
+@dataclass
+class TokenEndpointResponse:
+    access_token: str
+    token_type: str
+    scope: str
+    me: str
+
+
+def validate_authorization_response(
+    grant_type: str,
+    code: str,
+    client_id: str,
+    redirect_uri: str,
+    code_challenge: str,
+    code_challenge_method: str,
+    allowed_methods: list = ["S256"]
+) -> bool:
```

Comment:
As this will raise an exception with any error i.e. never returns `False`, do we need to return True or anything at all? If we want to return `True`, maybe we should rename the function to better match the return type? `is_authorization_response_valid` or so?